### ISSUE TITLE
self qualifiedByName get error: Qualifier error. No method found annotated with @Named#value

### ISSUE THREAD

**Issue by ileAuTresor** [mapstruct/mapstruct#3077]
### Expected behavior

Given a class that contains an attribute of the same class:

```
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class A {
    String id;
    A linkedA;
}

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ADto {
    private String id;
    private ADto linkedA;
}

@Mapper(componentModel = "spring",
        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
        injectionStrategy = InjectionStrategy.CONSTRUCTOR)
public interface AMapper {
    @Named("toADto")
    @Mapping(target = "linkedA", source = "linkedA", qualifiedByName = "toADto")
    ADto toDto(A a);
}
```

In 1.5.2.Final, the method `toDto` generated inside AMapperImpl can use recurive in itself
```
public ADto toDto(A a) {
        if ( a == null ) {
            return null;
        }
        ADto.ADtoBuilder aDto = ADto.builder();
        if ( a.getLinkedA() != null ) {
            aDto.linkedA( toDto( a.getLinkedA() ) );
        }

```

### Actual behavior

In 1.5.3.Final, an error is generated in compiled time: 
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project test: Compilation failure: Compilation failure: 
[ERROR] .../src/main/java/mapper/AMapper.java:[12,5] Qualifier error. No method found annotated with @Named#value: [ toADto ]. See https://mapstruct.org/faq/#qualifier for more info.
[ERROR] ...src/main/java/mapper/AMapper.java:[12,5] Can't map property "A linkedA" to "ADto linkedA". Consider to declare/implement a mapping method: "ADto map(A value)".

### Steps to reproduce the problem
Error on compilation with the following pom.xml: 
```

<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>2.7.5</version>
        <relativePath/> <!-- lookup parent from repository -->
    </parent>
    <groupId>org.example</groupId>
    <artifactId>test</artifactId>
    <version>1.0-SNAPSHOT</version>

    <properties>
        <maven.compiler.source>11</maven.compiler.source>
        <maven.compiler.target>11</maven.compiler.target>
        <lombok.version>1.18.24</lombok.version>
        <lombok-mapstruct-binding.version>0.2.0</lombok-mapstruct-binding.version>
        <mapstruct.version>1.5.3.Final</mapstruct.version>
    </properties>

    <dependencies>
        <!-- spring -->
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <!-- mapstruct -->
        <dependency>
            <groupId>org.mapstruct</groupId>
            <artifactId>mapstruct</artifactId>
            <version>${mapstruct.version}</version>
        </dependency>
        <dependency>
            <groupId>org.mapstruct</groupId>
            <artifactId>mapstruct-jdk8</artifactId>
            <version>${mapstruct.version}</version>
        </dependency>

        <!-- lombok -->
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <optional>true</optional>
            <version>${lombok.version}</version>
        </dependency>

    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <id>build-info</id>
                        <goals>
                            <goal>build-info</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>${maven-compiler-plugin.version}</version>
                <configuration>
                    <source>${java.version}</source>
                    <target>${java.version}</target>
                    <annotationProcessorPaths>
                        <path>
                            <groupId>org.mapstruct</groupId>
                            <artifactId>mapstruct-processor</artifactId>
                            <version>${mapstruct.version}</version>
                        </path>
                        <path>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok-mapstruct-binding</artifactId>
                            <version>${lombok-mapstruct-binding.version}</version>
                        </path>
                        <path>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok</artifactId>
                            <version>${lombok.version}</version>
                        </path>
                    </annotationProcessorPaths>
                    <useIncrementalCompilation>false</useIncrementalCompilation>
                </configuration>
            </plugin>
        </plugins>
    </build>
</project>

```

### MapStruct Version

1.5.3.Final

**Comment by Zegveld** [comment]
This should be fixed in the next release with #3057 that has already been merged.

**Comment by filiphr** [comment]
Indeed this will be fixed in the next release. I've anyways added a test case in PR #3080 since this is slightly different that #3057
